mlflow/projects/yarn.py

```diff
@@ -0,0 +1,288 @@
+            args=args,
+            name="MLflow run for experiment {}".format(experiment_id),
+            num_cores=yarn_config[YARN_NUM_CORES],
+            memory=yarn_config[YARN_MEMORY],
+            queue=yarn_config[YARN_QUEUE],
+            hadoop_file_systems=yarn_config[YARN_HADOOP_FILESYSTEMS].split(','),
+            hadoop_conf_dir=yarn_config[YARN_HADOOP_CONF_DIR],
+            env_vars=env,
+            additional_files=additional_files,
+            pex_env=pex_env
+        )
+
+        _logger.info("YARN backend launched app_id : %s", app_id)
+        return YarnSubmittedRun(skein_app_id=app_id, mlflow_run_id=remote_run.info.run_id)
+
+    raise ExecutionException("Not able to launch your job to YARN.")
+
+
+def _submit(skein_client, module_name, args=None, name="yarn_launcher",
+            num_cores=1, memory="1 GiB", pex_env=None,
+            hadoop_file_systems=None, hadoop_conf_dir="", queue=None, env_vars=None,
+            additional_files=None, node_label=None, num_containers=1,
+            user=None):
+    service = _generate_skein_service(memory, num_cores, num_containers, env_vars, additional_files,
+                                      hadoop_conf_dir, pex_env, module_name, args)
+
+    spec = skein.ApplicationSpec(
+        name=name,
+        file_systems=hadoop_file_systems,
+        services={name: service},
+        acls=skein.model.ACLs(
+            enable=True,
+            ui_users=['*'],
+            view_users=['*']
+        )
+    )
+
+    if user:
+        spec.user = user
+
+    if queue:
+        spec.queue = queue
+
+    if node_label:
+        service.node_label = node_label
+
+    return skein_client.submit(spec)
+
+
+def _generate_skein_service(memory, num_cores, num_containers, env_vars, additional_files,
+                            hadoop_conf_dir, pex_env, module_name, args):
+    env = dict(env_vars) if env_vars else dict()
+    env.update(
+        {
+            'SKEIN_CONFIG': './.skein',
+            'PEX_ROOT': "./.pex",
+            "PYTHONPATH": ".",
+            "GIT_PYTHON_REFRESH": "quiet"
+        }
+    )
+    launch_options = "-m" if not module_name.endswith(".py") else ""
+    launch_args = args if args else ""
+
+    dict_files_to_upload = {os.path.basename(path): os.path.abspath(path)
+                            if not _is_hdfs_path(path) else path for path in additional_files}
+    python_bin = "./%s" % os.path.basename(pex_env) if pex_env.endswith(
+        '.pex') else "./%s/bin/python" % os.path.basename(pex_env)
+
+    return skein.Service(
+        resources=skein.model.Resources(memory, num_cores),
+        instances=num_containers,
+        files=dict_files_to_upload,
+        env=env,
+        script="""
+                    set -x
+                    env
+                    export HADOOP_CONF_DIR=%s
+                    %s %s %s %s
+                """ % (hadoop_conf_dir, python_bin, launch_options, module_name, launch_args)
+    )
+
+
+def _is_hdfs_path(path):
```

Comment:
har ? 
I am pretty sure the prefix => hdfs are already registred in mlflow elsewhere. Could you reuse it.

(and BTW, you can even imagine to use yarn without hdfs)